### PR TITLE
[ET-1841] Feat - My Tickets page view links language tweaks.

### PR DESCRIPTION
**Pull request by rafsuntaskin**
### 🎫 Ticket

[ET-1841] 

### 🎥 Artifacts 
📷 screenshot(s):

**Before:**
Screenshot 2023-09-13 at 7 27 18 PM

**After:**
Screenshot 2023-09-13 at 7 13 31 PM

### ✔️ Checklist
- [ ] I've included a changelog entry in the readme.txt file. 
- [x] My code is tested. 
- [x] My code has [proper inline documentation](https://the-events-calendar.github.io/products-engineering/docs/code-standards/).

[ET-1841]: https://theeventscalendar.atlassian.net/browse/ET-1841?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/Tribe/Tickets_View.php [hunk 1]
```diff
@@ -811,19 +811,19 @@ public function has_ticket_attendees( $event_id, $user_id = null ) {
...
+	 * @since TBD Updated signature to accept RSVP and ticket count arguments and removed deprecated 3rd parameter.
 	 *
 	 * @param int      $event_id   The Event ID we're checking.
 	 * @param int|null $user_id    An optional User ID.
 	 * @param null     $deprecated Deprecated argument.
 	 *
 	 * @return string
 	 */
-	public function get_description_rsvp_ticket( $event_id, $user_id = null, $deprecated = null ) {
+	public function get_description_rsvp_ticket( $event_id, $user_id = null, $rsvp_count = null, $ticket_count = null ) {
```
Comment (lucatume) [inline src/Tribe/Tickets_View.php:822]: Here you add one argument per RSVP count, one for Tickets' count: how will this scale with more type of tickets?

It looks like you're not using this function in the modified code: why change it?

File: src/views/blocks/attendees/view-link.php [hunk 2]
```diff
@@ -47,33 +48,50 @@
...
-}
+$link       = $view->get_tickets_page_url( $event_id, $is_event_page );
+$link_label = $rsvp_count > 0 && $ticket_count > 0 ? __( 'View all' ) : __( 'View ', 'event-tickets' );
+
+if ( $rsvp_count > 0 ) {
+	// Translators: 1: the number of RSVPs, 2: singular RSVP label, 3: plural RSVP label.
+	$counters[] = sprintf(
+		_n( '%1$d %2$s', '%1$d %3$s', $rsvp_count, 'event-tickets' ),
+		$rsvp_count,
+		tribe_get_rsvp_label_singular( basename( __FILE__ ) ),
```
Comment (lucatume) [inline src/views/blocks/attendees/view-link.php:63]:
```suggestion
		tribe_get_rsvp_label_singular( 'attendees-view-link' ),
```
Using a statically readable, thus searchable, context line will make understanding the context easier.

File: src/views/blocks/attendees/view-link.php [hunk 3]
```diff
@@ -47,33 +48,50 @@
...
+		tribe_get_rsvp_label_plural( basename( __FILE__ )
```
Comment (lucatume) [inline src/views/blocks/attendees/view-link.php:64]: Ditto for this context:
```suggestion
		tribe_get_rsvp_label_plural( 'attendees-view-link's
```
Also pull up the closing parenthesis.

File: src/views/blocks/attendees/view-link.php [hunk 4]
```diff
@@ -47,33 +48,50 @@
...
+		)
+	);
 
-if ( empty( $counters ) ) {
-	return false;
+	// Append label on link.
+	if ( empty( $ticket_count ) ) {
+		$link_label .= _n( tribe_get_rsvp_label_singular(), tribe_get_rsvp_label_plural(), $rsvp_count, 'event-tickets' );
```
Comment (lucatume) [inline src/views/blocks/attendees/view-link.php:70]: Provide the context to `tribe_get_rsvp_label_singular` and `tribe_get_rsvp_label_plural` here?

File: src/views/blocks/attendees/view-link.php [hunk 5]
```diff
@@ -47,33 +48,50 @@
...
+	}
 }
 
-$link = $view->get_tickets_page_url( $event_id, $is_event_page );
+if ( $ticket_count > 0 ) {
+	// Translators: 1: the number of Tickets, 2: singular Ticket label, 3: plural Ticket label.
+	$counters[] = sprintf(
+		_n( '%1$d %2$s', '%1$d %3$s', $ticket_count, 'event-tickets' ),
+		$ticket_count,
+		tribe_get_ticket_label_singular( basename( __FILE__ ) ),
```
Comment (lucatume) [inline src/views/blocks/attendees/view-link.php:79]: Consideration about the context applies here and to the following line as well.

File: src/views/blocks/attendees/view-link.php [hunk 6]
```diff
@@ -47,33 +48,50 @@
...
+		tribe_get_ticket_label_plural( basename( __FILE__ )
+		)
+	);
+
+	// Append label on link.
+	if ( empty( $rsvp_count ) ) {
+		$link_label .= _n( tribe_get_ticket_label_singular(), tribe_get_ticket_label_plural(), $ticket_count, 'event-tickets' );
```
Comment (lucatume) [inline src/views/blocks/attendees/view-link.php:86]: Here too: add a context.

File: src/views/tickets/view-link.php [hunk 7]
```diff
@@ -21,92 +21,16 @@
...
-	<a href="<?php echo esc_url( $link ); ?>">
-		<?php
-		// Translators: %s: The name(s) of the type(s) of ticket(s) the specified user (optional) has for the specified event.
-		echo sprintf( esc_html__( 'View your %s', 'event-tickets' ), $view->get_description_rsvp_ticket( $event_id, $user_id ) );
-		?>
-	</a>
-</div>
+/** @var Tribe__Tickets__Editor__Template $template */
+$template = tribe( 'tickets.editor.template' );
+$template->template( 'blocks/attendees/view-link' );
```
Comment (lucatume) [inline src/views/tickets/view-link.php:36]: Loading this file, and trigger all the attached filtering, might be skipped simply including the `blocks/attendees/view-link` template directly in the file that would include this one. Why keep including this file?

File: tests/integration/Tribe/Tickets/MyTickets/MyTicketsTest.php [hunk 8]
```diff
@@ -0,0 +1,227 @@
...
+use Closure;
+use Codeception\TestCase\WPTestCase;
+use Generator;
+use tad\Codeception\SnapshotAssertions\SnapshotAssertions;
+use Tribe\Tests\Traits\With_Uopz;
+use Tribe\Tickets\Test\Commerce\Attendee_Maker;
+use Tribe\Tickets\Test\Commerce\TicketsCommerce\Ticket_Maker;
+use Tribe\Tickets\Test\Commerce\RSVP\Ticket_Maker as RSVP_Ticket_Maker;
+
+class MyTicketsTest extends WPTestCase {
```
Comment (lucatume) [inline tests/integration/Tribe/Tickets/MyTickets/MyTicketsTest.php:14]: Excellent.

**Review by lucatume — CHANGES_REQUESTED** [review]
Some questions about the implementation, some smaller change, otherwise the PR is 90% spot-on. Tests are excellent.

File: src/Tribe/Tickets_View.php (hunk 1 above)
Comment (rafsuntaskin) [reply src/Tribe/Tickets_View.php:822]: Yes, you are correct. I have skipped using this function within the template as I found the function name is specific to `rsvp` but used in the ticket context as well. 

The changes were made before I updated the template. Now, they don't have any impact on the changes that are made from this PR. I was thinking of reverting it as well. But I kept it in place as separating the dependency is still a kind of improvement from the existing one.

We won't need to scale it, as we are now building the labels within the template, as that feels more logical. 

This method is now only used here: https://github.com/the-events-calendar/event-tickets/blob/d2b8d1aac35c9ed6c30531758ee09317e650ba4e/src/views/tickets/orders.php#L169

Ideally, we should move everything out of the template and then set as template vars, but the same template is used for legacy views as well, so I didn't want to increase the scope of the original changes within this PR.

File: src/views/tickets/view-link.php (hunk 7 above)
Comment (rafsuntaskin) [reply src/views/tickets/view-link.php:36]: As discussed above, this template is a legacy view template. It was just doing the same thing. But as we can not simply remove and replace the template to keep backward compatibility ( if overridden from the theme ), I just re-used the new template in it.

File: src/Tribe/Tickets_View.php (hunk 1 above)
Comment (rafsuntaskin) [reply src/Tribe/Tickets_View.php]: I have reverted the changes, made to this one. I think it makes more sense to skip updating it as we are not using it within this PR.

**Review by lucatume — APPROVED** [review]

**Pull request merged**